packages/commerce-sdk-react/src/hooks/useAsync.ts

```diff
@@ -4,25 +4,44 @@
  * SPDX-License-Identifier: BSD-3-Clause
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
-import {useState} from 'react'
+import {useState, useEffect} from 'react'
 import {ActionResponse, QueryResponse} from './types'
 
-export const useAsync = <T>(fn: () => Promise<T>, deps?: unknown[]): QueryResponse<T> => {
-    // This is a stub implementation to validate the types.
-    // The real implementation will be more React-y.
-    const result: QueryResponse<T> = {
-        isLoading: true
-    }
-    fn()
-        .then((data) => {
-            result.isLoading = false
-            result.data = data
-        })
-        .catch((error) => {
-            result.isLoading = false
-            result.error = error
-        })
-    return result
+export const useAsync = <T>(fn: () => Promise<T>, deps: unknown[] = []): QueryResponse<T> => {
```

Comment:
This is a rough implementation for this hook as it is outside of the scope of my current ticket, I believe we need another ticket to make this hook use `useServeEffect` for server-side rendering, and use `useSWR` for client side rendering